doc/nrf/libraries/networking/rest_client.rst

```diff
@@ -0,0 +1,45 @@
+.. _lib_rest_client:
+
+REST client
+###########
+
+.. contents::
+   :local:
+   :depth: 2
+
+The REST client library provides means for performing REST requests.
+
+Overview
+********
+
+This library creates a socket with TLS when requested.
+It uses Zephyr's HTTP client library to send HTTP requests and receive HTTP responses.
+When using this library, the :c:struct:`rest_client_req_resp_context` structure is populated and passed to the :c:func:`rest_client_request` function.
+The same structure will contain the response data.
+
+Configuration
+*************
+
+To use the REST client library, enable the :kconfig:`CONFIG_REST_CLIENT` Kconfig option.
+
+You can configure the following options to adjust the behavior of the library:
+
+*  :kconfig:`CONFIG_REST_CLIENT_REQUEST_TIMEOUT`
+*  :kconfig:`CONFIG_REST_CLIENT_SCKT_SEND_TIMEOUT`
+*  :kconfig:`CONFIG_REST_CLIENT_SCKT_RECV_TIMEOUT`
+*  :kconfig:`CONFIG_REST_CLIENT_SCKT_TLS_SESSION_CACHE_IN_USE`
+
+Limitations
+***********
+
+*  Executing REST request is a blocking operation.
```

Comment:
```suggestion
Executing a REST request is a blocking operation.
The application thread is blocked until until the request has completed.
```

Reply:
I will add the following:
`The calling thread is blocked until the request has completed.`